From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch 05-04

  Description: kernel: Reported memory size is wrong.
  Symptom:     Reported memory size wrong or system hangs on IPL.
  Problem:     The wrong base register is used to read a value from data
               structure. The value is used to calculate the memory size.
  Solution:    Use correct register.
  Problem-ID:  24316
  Archs:       s390-31, s390-64

Acked-by: Ihno Krumreich <ihno@suse.de>

--- linux-2.5/arch/s390/kernel/head31.S	14 Nov 2005 16:22:34 -0000	1.2
+++ linux-2.5/arch/s390/kernel/head31.S	1 Jun 2006 12:45:10 -0000	1.2.4.1
@@ -87,7 +87,7 @@
 	ic	%r3,.Lscpa1-PARMAREA(%r4)
 	chi	%r3,0x00
 	jne	.Lcompmem
-	l	%r3,.Lscpa2-PARMAREA(%r13)
+	l	%r3,.Lscpa2-PARMAREA(%r4)
 .Lcompmem:
 	mr	%r2,%r1			# mem in MB on 128-bit
 	l	%r1,.Lonemb-.LPG1(%r13)

--- linux-2.5/arch/s390/kernel/head64.S	29 Oct 2005 10:57:51 -0000	1.22
+++ linux-2.5/arch/s390/kernel/head64.S	1 Jun 2006 12:45:10 -0000	1.22.4.1
@@ -86,7 +86,7 @@
 	ic    %r3,.Lscpa1-PARMAREA(%r4)
 	chi   %r3,0x00
 	jne   .Lcompmem
-	l     %r3,.Lscpa2-PARMAREA(%r13)
+	l     %r3,.Lscpa2-PARMAREA(%r4)
 .Lcompmem:
 	mlgr  %r2,%r1			# mem in MB on 128-bit
 	l     %r1,.Lonemb-.LPG1(%r13)

